flytekit/image_spec/default_builder.py

```diff
@@ -216,35 +221,48 @@ def prepare_python_install(image_spec: ImageSpec, tmp_dir: Path) -> str:
         extra_urls = [f"--extra-index-url {url}" for url in image_spec.pip_extra_index_url]
         pip_install_args.extend(extra_urls)
 
+    pip_secret_mount = ""
+    pip_secret_mount = " ".join(
+        f"--mount=type=secret,id={_secret_id(i)},target={dst}"
+        for i, (_, dst) in enumerate(image_spec.pip_secret_mounts or [])
+    )
```

Comment:
Given the definition of `pip_secret_mount = ""` above, I think checking for `pip_secret_mounts` was the intention?

```suggestion
    if self.pip_secret_mounts:
        pip_secret_mount = " ".join(
            f"--mount=type=secret,id={_secret_id(i)},target={dst}"
            for i, (_, dst) in enumerate(image_spec.pip_secret_mounts)
        )
```

Reply:
The proposal LGTM. This was based on flytebot suggestion.
https://github.com/flyteorg/flytekit/pull/3082/commits/eab741f82c79ef4e78eb5107ddcc2aed58ff3829